<a href="https://colab.research.google.com/github/novalsakti02-kibo/Pemerograman-dasar/blob/main/ats_praktikum__kelompok_8_pemrograman_dasar.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# 1. Percabangan / Branching

# Meminta input suhu dari pengguna
suhu = float(input("Masukkan suhu air pendingin (°C): "))

# Mengecek kondisi suhu
if suhu <= 90:
    print("STATUS: Normal")
elif suhu > 90 and suhu < 100:
    print("STATUS: Warning")
else:
    print("ALERT: SHUTDOWN")



Masukkan suhu air pendingin (°C): 96
STATUS: Warning


 Penjelasan Program 1

Program ini dibuat untuk memantau suhu air pendingin pada sistem pembangkit listrik. Saat dijalankan, program akan meminta pengguna memasukkan nilai suhu dalam satuan °C, kemudian data tersebut disimpan dalam variabel suhu menggunakan float(input(...)) agar dapat menerima angka desimal.

Setelah itu, program melakukan pengecekan kondisi dengan struktur if-elif-else.

- Jika suhu yang dimasukkan suhunya 90°C atau kurang, program bakal nampilin "STATUS: Normal". Artinya sistem masih aman dan nggak ada masalah.

- Kalau suhunya lebih dari 90°C tapi masih di bawah 100°C, program bakal ngasih info "STATUS: Warning". Ini tanda kalau suhu mulai naik dan perlu diperhatikan.

- Kalau suhunya 100°C atau lebih, program langsung nampilin "ALERT: SHUTDOWN", yang berarti kondisinya udah bahaya dan sistem harus segera dimatikan biar nggak rusak atau terjadi kegagalan.

Secara singkat, program ini bisa bantu operator atau teknisi buat memantau suhu pendingin. Jadi mereka bisa tau kapan kondisi normal, kapan harus waspada, dan kapan harus langsung ambil tindakan. Dengan begitu, risiko kerusakan sistem karena suhu terlalu tinggi bisa diminimalisir. Program ini intinya sebagai sistem pemantauan awal biar operasi pembangkit tetap aman.

In [ ]:
# 2. Fungsi / Functions

# Fungsi untuk menghitung efisiensi turbin
def hitung_efisiensi(input_energi, output_energi):
    efisiensi = output_energi / input_energi
    return efisiensi

# Contoh pemanggilan fungsi (Bonus)
hasil = hitung_efisiensi(500, 400)
print("Efisiensi Turbin:", hasil)


Efisiensi Turbin: 0.8


Penjelasan Program 2

Program ini dibuat untuk menghitung efisiensi turbin pada sistem pembangkit menggunakan sebuah fungsi bernama hitung_efisiensi. Fungsi ini bekerja dengan menerima dua parameter utama, yaitu input_energi dan output_energi. Proses di dalam program dapat dijelaskan sebagai berikut:

- input_energi merepresentasikan energi yang masuk ke turbin.

- output_energi adalah energi yang dihasilkan oleh turbin.

- Di dalam fungsi, efisiensi dihitung menggunakan rumus efisiensi = output_energi / input_energi, sesuai konsep dasar konversi energi.

- Nilai efisiensi kemudian dikembalikan menggunakan return, sehingga bisa digunakan atau ditampilkan di luar fungsi.

- Sebagai contoh pemanggilan fungsi, diberikan nilai input_energi = 500 dan output_energi = 400. Hasil perhitungan disimpan dalam variabel hasil, lalu ditampilkan dengan perintah print.

Dengan program ini, kita dapat mengetahui seberapa baik turbin mengubah energi masuk menjadi energi keluaran, sehingga sangat membantu dalam menganalisis kinerja sistem pembangkit secara efisien dan tepat.

In [ ]:
# 3. Looping for

vibrasi_mesin = [1.2, 0.9, 2.5, 3.1, 1.8]

for i in range(len(vibrasi_mesin)):
    if vibrasi_mesin[i] > 2.0:
        print(f"Warning: Unit {i+1} vibrasi tinggi!")

Penjelasan Program 3

Program ini dibuat untuk mengecek getaran dari 5 mesin, yang di mana datanya itu disimpan di list bernama vibrasi_mesin. Nilai yang ada di dalam list dia menunjukkan seberapa besar getaran di tiap unit mesinnya. Program ini menggunakan perulangan for supaya bisa membaca satu persatu data dari semua mesin menggunakan indeks.

Di dalam perulangan ada kondisi if yang berfungsi untuk memeriksa apakah nilai getaran lebih dari 2.0. jika nilainya lebih maka program akan menampilkan pesan "Warning: Unit [nomor_unit] vibrasi tinggi!". Di mana nomor unik ditampilkan mulai dari angka 1 supaya lebih mudah dibaca. Dengan program ini kita bisa mengetahui mesin mana yang getarannya itu terlalu tinggi, sehingga bisa langsung kita cek atau perbaiki sebelum terjadinya kerusakan pada mesin tersebut.

In [ ]:
#4. Logika if Majemuk/bertingkat

tekanan_uap = float(input("Masukkan tekanan uap (psi): "))
status_manual_override = False  # Misal override dalam keadaan OFF

# Logika if majemuk/bertingkat
if tekanan_uap > 150 and status_manual_override == False:
    print("Katup Terbuka")
else:
    print("Katup Tetap Tertutup")


Masukkan tekanan uap (psi): 151
Katup Terbuka


Penjelasan Program 4

Program ini digunakan untuk menentukan apakah sebuah katup pengaman (safety valve) boleh terbuka berdasarkan kondisi tekanan uap dan status pengaturan manual. Dalam program ini, terdapat dua variabel utama, yaitu tekanan_uap yang bertipe float sebagai nilai tekanan uap dalam satuan psi, dan status_manual_override yang bertipe boolean untuk menunjukkan apakah katup sedang di-override secara manual atau tidak. Logika program menggunakan operator logika and, sehingga kedua syarat harus terpenuhi agar katup dapat terbuka. Kondisinya adalah sebagai berikut:

- Tekanan uap harus lebih besar dari 150 psi.

- status_manual_override harus bernilai False, artinya tidak ada override


Jika kedua kondisi tersebut benar, maka perintah print("Katup Terbuka") akan dijalankan, menandakan bahwa sistem mengizinkan katup terbuka secara otomatis. Namun jika salah satu atau kedua kondisi tidak terpenuhi, maka program akan mengeksekusi bagian else dan mencetak “Katup Tetap Tertutup”. Dengan logika ini, program membantu memastikan sistem keamanan bekerja dengan benar, hanya membuka katup ketika tekanan berbahaya dan tidak ada intervensi manual yang menahan katup tetap tertutup.

In [ ]:
# 5. Looping while

# Inisialisasi suhu awal
suhu_saat_ini = 60

# Loop akan berjalan selama suhu masih di bawah 85°C
while suhu_saat_ini < 85:
    print("Suhu saat ini:", suhu_saat_ini, "°C")
    # Naikkan suhu sebesar 5°C tiap iterasi
    suhu_saat_ini += 5

# Setelah loop selesai
print("Suhu telah mencapai atau melebihi 85°C.")

Suhu saat ini: 60 °C
Suhu saat ini: 65 °C
Suhu saat ini: 70 °C
Suhu saat ini: 75 °C
Suhu saat ini: 80 °C
Suhu telah mencapai atau melebihi 85°C.


Penjelasan Program 5

Program ini dibuat untuk mensimulasikan proses pemanasan suhu menggunakan perulangan while, di mana sistem akan menaikkan suhu secara bertahap hingga mencapai suhu target tertentu. Proses di dalam program dapat dijelaskan sebagai berikut:
- Inisialisasi suhu awal: Variabel suhu_saat_ini diatur bernilai 60°C, yang merepresentasikan suhu awal sistem sebelum pemanasan dimulai.
- Kondisi perulangan: Program menggunakan perintah while suhu_saat_ini < 85: agar proses pemanasan berjalan terus selama suhu belum mencapai 85°C.

- Isi perulangan: Di dalam blok perulangan, program menampilkan suhu saat ini dengan perintah print("Suhu saat ini:", suhu_saat_ini, "°C"), kemudian menaikkan nilai suhu sebesar 5°C tiap iterasi menggunakan suhu_saat_ini += 5.

- Berhenti ketika kondisi tidak terpenuhi: Jika suhu_saat_ini sudah mencapai atau melebihi 85°C, maka kondisi suhu_saat_ini < 85 menjadi False, sehingga perulangan berhenti secara otomatis.

- Setelah perulangan selesai, program menampilkan pesan Suhu telah mencapai atau melebihi 85°C. sebagai tanda bahwa proses pemanasan telah selesai.

Dengan program ini, kita dapat mensimulasikan kenaikan suhu secara bertahap menggunakan logika perulangan, yang menggambarkan bagaimana sistem pemanas bekerja hingga mencapai suhu target secara otomatis dan efisien.

In [ ]:
# 6. Variabel & Ekspresi

# Program menghitung daya listrik generator
tegangan = 220   # Volt
arus = 15        # Ampere

# Rumus daya listrik
daya = tegangan * arus

# Menampilkan hasil
print("Daya listrik yang dihasilkan generator adalah:", daya, "Watt")


Daya listrik yang dihasilkan generator adalah: 3300 Watt


Program ini digunakan untuk menghitung daya listrik yang dihasilkan oleh sebuah generator berdasarkan nilai tegangan dan arus yang diketahui. Pada program ini terdapat dua variabel utama, yaitu tegangan yang bernilai 220 Volt dan arus yang bernilai 15 Ampere. Kedua nilai ini kemudian digunakan dalam rumus perhitungan daya listrik, yaitu P = V × I, di mana P adalah daya, V adalah tegangan, dan I adalah arus. Program mengalikan nilai tegangan dengan arus dan menyimpan hasilnya dalam variabel daya. Setelah perhitungan selesai, program menampilkan hasil daya listrik melalui perintah print, yaitu sebesar 3300 Watt. Dengan program sederhana ini, kita dapat mengetahui berapa besar daya yang dihasilkan generator berdasarkan parameter tegangan dan arus yang diberikan, sehingga program ini mendukung pemahaman perhitungan dasar dalam sistem pembangkit listrik.

In [ ]:
# 7. Percabangan / Branching

# Program untuk mengklasifikasikan beban listrik

# Input data dari pengguna
kapasitas_maks = float(input("Masukkan kapasitas maksimum (Watt): "))
beban_terpakai = float(input("Masukkan beban terpakai (Watt): "))

# Hitung persentase pemakaian
persentase = (beban_terpakai / kapasitas_maks) * 100

# Klasifikasi berdasarkan persentase
if persentase <= 60:
    status = "Beban Ringan"
elif persentase <= 85:
    status = "Beban Sedang"
else:
    status = "Beban Berat"

# Tampilkan hasil
print(f"Persentase pemakaian: {persentase:.2f}%")
print("Status beban:", status)

Penjelasan Program 7

Program ini dibuat untuk mengklasifikasikan beban listrik berdasarkan persentase pemakaian terhadap kapasitas maksimum menggunakan struktur percabangan (if-elif-else).

Proses di dalam program dapat dijelaskan sebagai berikut:

- Penerimaan input pengguna: Program meminta pengguna memasukkan dua nilai, yaitu kapasitas_maks (kapasitas maksimum daya listrik) dan beban_terpakai (daya listrik yang sedang digunakan).

- Perhitungan persentase pemakaian: Persentase dihitung menggunakan rumus: persentase = (beban_terpakai / kapasitas_maks) * 100 Rumus ini menunjukkan seberapa besar beban terpakai dibandingkan dengan kapasitas maksimum sistem.

- Struktur percabangan: Program menggunakan if-elif-else untuk menentukan status beban: -Jika persentase ≤ 60, maka status “Beban Ringan”. -Jika persentase > 60 dan ≤ 85, maka status “Beban Sedang”. -Jika persentase > 85, maka status “Beban Berat”.

- Menampilkan hasil: Setelah klasifikasi selesai, program akan menampilkan hasil berupa persentase pemakaian dan status beban sesuai kondisi yang dihitung.

- Contoh hasil eksekusi: Jika pengguna memasukkan kapasitas_maks = 1000 dan beban_terpakai = 750, maka persentase pemakaian = 75%, dan program menampilkan status “Beban Sedang”.

Dengan program ini, pengguna dapat mengetahui tingkat penggunaan daya listrik secara efisien. Hal ini berguna untuk memantau kondisi sistem listrik agar tidak melebihi kapasitas maksimum dan tetap bekerja secara aman.

In [ ]:
# 8. Fungsi / Modularisasi

# Fungsi menghitung efisiensi termal turbin gas
def hitung_efisiensi_termal():
    # Input energi dari pengguna
    energi_masuk = float(input("Masukkan energi masuk (Joule): "))
    energi_keluar = float(input("Masukkan energi keluar (Joule): "))

    # Perhitungan efisiensi
    efisiensi = (energi_keluar / energi_masuk) * 100

    # Menampilkan hasil dengan format rapi
    print("\n===== HASIL PERHITUNGAN EFISIENSI =====")
    print(f"Energi Masuk       : {energi_masuk} Joule")
    print(f"Energi Keluar      : {energi_keluar} Joule")
    print(f"Efisiensi Termal   : {efisiensi:.2f}%")
    print("========================================")

# Memanggil fungsi
hitung_efisiensi_termal()



Masukkan energi masuk (Joule): 500
Masukkan energi keluar (Joule): 225

===== HASIL PERHITUNGAN EFISIENSI =====
Energi Masuk       : 500.0 Joule
Energi Keluar      : 225.0 Joule
Efisiensi Termal   : 45.00%


Penjelasan Program 8

Program ini digunakan untuk menghitung efisiensi termal pada turbin gas berdasarkan perbandingan antara energi masuk dan energi keluar. Di dalam fungsi hitung_efisiensi_termal(), pengguna akan diminta memasukkan dua nilai, yaitu energi_masuk dan energi_keluar, yang masing-masing bertipe float karena nilai energi dapat berupa bilangan desimal. Setelah nilai dimasukkan, program menghitung efisiensi menggunakan rumus Efisiensi (%) = (energi keluar / energi masuk) × 100. Hasil perhitungan kemudian ditampilkan dengan format yang rapi, menampilkan energi masuk, energi keluar, serta nilai efisiensi yang sudah diformat dua angka di belakang koma.

Sebagai contoh, jika nilai energi masuk = 500 Joule dan energi keluar = 225 Joule, maka perhitungannya adalah:

- Efisiensi = (225 / 500) × 100

- Efisiensi = 0,45 × 100

- Efisiensi = 45%

Dengan demikian, program ini membantu dalam memahami efisiensi termal suatu turbin gas, yaitu seberapa besar energi yang berhasil dikonversi menjadi energi keluaran dibandingkan energi yang masuk, sehingga dapat digunakan sebagai parameter evaluasi performa sistem pembangkitan.

In [ ]:
# 9. Looping + List

suhu_boiler = [92.5, 88.7, 95.1, 102.3, 97.8]

for i in range(len(suhu_boiler)):
    print(f"Titik {i+1}: {suhu_boiler[i]}°C")
    if suhu_boiler[i] > 100:
        print(f"Peringatan: Suhu di titik {i+1} melebihi 100°C")

Titik 1: 92.5°C
Titik 2: 88.7°C
Titik 3: 95.1°C
Titik 4: 102.3°C
Peringatan: Suhu di titik 4 melebihi 100°C
Titik 5: 97.8°C


Penjelasan Program 9

Pada program ini saya membuat program untuk menampilkan hasil pembacaan suhu dari 5 titik di boiler. Di mana semua data suhunya itu disimpan di dalam list bernama suhu_boiler. Di dalam program, saya menggunakan perulangan for supaya programnya bisa menampilkan suhu dari setiap titik satu persatu.

Setelah suhu ditampilkan, program akan mengecek apakah ada suhu yang melebihi 100°C dengan menggunakan kondisi if. Kemudian jika ada suhu di titik tertentu lebih dari 100°C, maka programnya akan langsung menampilkan pesan peringatan bahwa suhu di titik boiler tersebut terlalu tinggi. Dari program ini bisa kita gunakan untuk menentukan di mana titik yang suhunya itu melewati batas aman, sehingga bisa dicek dan ditangani supaya boilernya tetap bekerja dengan baik dan aman.

In [ ]:
# 10. While Loop & Logika

# Simulasi pengisian tangki bahan bakar

tangki = 0  # mulai dari 0 liter

while tangki <= 100:
    print(f"Volume bahan bakar saat ini: {tangki} liter")

    # Kondisi hampir penuh
    if tangki == 80:
        print("WARNING: Hampir penuh!")

    # Kondisi penuh
    if tangki == 100:
        print("TANGKI PENUH – Pengisian berhenti.")
        break  # menghentikan loop

    tangki += 10  # menambah 10 liter setiap iterasi


Volume bahan bakar saat ini: 0 liter
Volume bahan bakar saat ini: 10 liter
Volume bahan bakar saat ini: 20 liter
Volume bahan bakar saat ini: 30 liter
Volume bahan bakar saat ini: 40 liter
Volume bahan bakar saat ini: 50 liter
Volume bahan bakar saat ini: 60 liter
Volume bahan bakar saat ini: 70 liter
Volume bahan bakar saat ini: 80 liter
Volume bahan bakar saat ini: 90 liter
Volume bahan bakar saat ini: 100 liter
TANGKI PENUH – Pengisian berhenti.


Penjelasan Program 10

Program ini digunakan untuk mensimulasikan proses pengisian tangki bahan bakar mulai dari kondisi kosong hingga penuh dengan kapasitas maksimum 100 liter. Variabel tangki diawali dengan nilai 0 liter sebagai kondisi awal. Selanjutnya, program menggunakan perulangan while yang berjalan selama nilai tangki kurang dari atau sama dengan 100. Pada setiap iterasi, program menampilkan volume bahan bakar saat ini kemudian menambahkan 10 liter menggunakan tangki += 10, sehingga level bahan bakar meningkat secara bertahap.

  
Di dalam loop terdapat dua kondisi pengecekan. Pertama, ketika volume mencapai 80 liter, program mencetak pesan "WARNING: Hampir penuh!" sebagai tanda bahwa kapasitas hampir maksimal dan proses pengisian perlu diperhatikan. Kedua, saat volume mencapai 100 liter, program menampilkan pesan "TANGKI PENUH – Pengisian berhenti.", yang berarti tangki sudah mencapai kapasitas maksimum dan proses pengisian harus dihentikan. Pada kondisi ini, perintah break digunakan untuk menghentikan loop. Dengan simulasi sederhana ini, kita dapat memahami bagaimana logika perulangan dan pengecekan kondisi dapat diterapkan dalam sistem monitoring pengisian bahan bakar.

In [ ]:
# Challenge #

# ============================================
# Sistem Monitoring Energi PLTS Mini-Grid
# Menghitung produksi energi harian panel surya
# dan memantau status baterai
# ============================================

# Fungsi menghitung total energi panel surya
def hitung_total_energi(energi_per_jam):
    total = 0
    for energi in energi_per_jam:  # perulangan for
        total += energi
    return total

# Fungsi mengecek status baterai
def cek_status_baterai(kapasitas):
    if kapasitas > 80:
        return "Status: ✅ Baterai Penuh (Sistem Aman)"
    elif kapasitas > 40:
        return "Status: ⚡ Baterai Cukup (Normal)"
    elif kapasitas > 20:
        return "Status: 🟡 Baterai Rendah (Perlu Perhatian)"
    else:
        return "Status: 🔴 KRITIS! Segera Isi / Aktifkan Backup"

print("===== SISTEM MONITORING PLTS MINI-GRID =====")

# Input energi surya setiap jam (5 jam simulasi puncak matahari)
energi_jam = []
jam = 1

while jam <= 5:  # loop pengumpulan data
    energi = float(input(f"Masukkan energi panel jam ke-{jam} (kWh): "))
    energi_jam.append(energi)
    jam += 1

# Hitung total energi harian
total_energi = hitung_total_energi(energi_jam)
print(f"\nTotal Energi Surya Hari Ini: {total_energi:.2f} kWh")

# Input kapasitas baterai PLTS
kapasitas = float(input("Masukkan kapasitas baterai saat ini (%): "))
status_baterai = cek_status_baterai(kapasitas)

# Output Status
print("\n===== STATUS SISTEM =====")
print(f"Kapasitas Baterai: {kapasitas}%")
print(status_baterai)

# Logika operasional berdasarkan kondisi
if kapasitas <= 20:
    print("Rekomendasi: 🔥 Aktifkan genset cadangan / charging cepat")
elif total_energi < 15:
    print("Catatan: 🌥️ Produksi energi rendah, cuaca tidak optimal")
else:
    print("Sistem Operasional: ✅ Stabil & Optimal")


===== SISTEM MONITORING PLTS MINI-GRID =====
Masukkan energi panel jam ke-1 (kWh): 3.2
Masukkan energi panel jam ke-2 (kWh): 3.5
Masukkan energi panel jam ke-3 (kWh): 4.1
Masukkan energi panel jam ke-4 (kWh): 3.8
Masukkan energi panel jam ke-5 (kWh): 3.4

Total Energi Surya Hari Ini: 18.00 kWh
Masukkan kapasitas baterai saat ini (%): 65

===== STATUS SISTEM =====
Kapasitas Baterai: 65.0%
Status: ⚡ Baterai Cukup (Normal)
Sistem Operasional: ✅ Stabil & Optimal


Penjelasan Program Challenge

Program ini mensimulasikan sistem monitoring untuk Pembangkit Listrik Tenaga Surya (PLTS) skala mini-grid. Tujuannya adalah memantau produksi energi harian dari panel surya sekaligus mengecek status baterai penyimpanan, yang merupakan komponen penting pada sistem energi terbarukan modern.

Pertama, program menggunakan while loop untuk mengambil input energi listrik yang dihasilkan panel surya setiap jam selama periode puncak matahari (5 jam). Data setiap jam disimpan dalam list. Selanjutnya, fungsi hitung_total_energi() yang menggunakan for loop menjumlahkan seluruh energi yang dihasilkan untuk mendapatkan total produksi energi harian.

Program juga memiliki fungsi cek_status_baterai() yang menggunakan logika if-elif-else untuk menentukan kondisi baterai berdasarkan persentasenya, mulai dari aman sampai pada kondisi kritis. Setelah mendapatkan input baterai dari pengguna, program mencetak status baterai dan memberikan rekomendasi operasional sistem. Jika baterai di bawah 20%, sistem menyarankan aktivasi genset cadangan atau pengisian cepat. Jika produksi energi surya rendah, program memberi catatan kondisi cuaca kurang optimal. Jika semuanya normal, sistem dinyatakan stabil.

Program ini mencerminkan kondisi nyata sistem energi terbarukan berbasis PLTS, terutama pada monitoring harian dan manajemen energi untuk efisiensi dan keandalan sistem.